### PR TITLE
Introducing Bounty Hub, a central hub to import and manage your hackerone programs to reNgine

### PR DESCRIPTION
**Pull request by null-ref-0000**
#1408

### CODE REVIEW

**Comment by yogeshojha** [comment]
@null-ref-0000 Excellent PR. I reviewed your PR and we need to make some changes. You do not have to worry as I will be making the required changes here is what I will do.

1. The endpoints are authenticated endpoints, likely we need api keys from hackerone. Though we have a hackerone key input in report section but that is for different purpose for reporting, so we will have to take this api input to API Vault section. This includes for both btw, we will have one input for hackerone API key which will be utilized by both reporting and syncing the bookmarked organizations.

2. I will also add Sync option under Quick Add option for us to be able to manually sync the targets

3. Since we will be utilizing the hackerone api already, it is a better idea for user to be able to import targets not only from bookmarked ones but we should let them search the target and import. 
For example while adding the target we can have a search box that will search and import targets, we should let user's which targets to import.

4. We will take the logic out of celery as it will be an overkill for this feature, I will put it inside api/views

I will take care of this changes today.

Rest, I must say an excellent feature and amazing PR! Loved it.

if you can help me with bugcrowd as well, let me know I havent used bugcrowd yet, or maybe even other bug bounty platforms like intigriti, see if we can use them somehow with this feature.

File: docker-compose.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ services:
       - tool_config:/root/.config
       - static_volume:/usr/src/app/staticfiles/
     environment:
-      - DEBUG=0
```
Comment (yogeshojha) [inline docker-compose.yml:40]: nooo ;p

**Comment by yogeshojha** [comment]
Unnecessary changes have been now removed, we can consider this as an excellent base to start with.

File: docker-compose.yml (hunk 1 above)
Comment (null-ref-0000) [reply docker-compose.yml]: my bad, forgot to revert this back after troubleshooting.

**Comment by null-ref-0000** [comment]
I'll start working on similar features for bugcrowd and intigriti. Will submit individual feature requests and PRs. 


"We will take the logic out of celery as it will be an overkill for this feature, I will put it inside api/views"

If we expand this won't it be useful for this to run as background job ? When building it, I had it in the api/views but thought maybe was better as background job as we reaching out to external APIs. The H1 APIs are fast, but as this feature expands the time to synchronize could grow and thus user waiting on the UI might be a bad experience ?   


"Since we will be utilizing the hackerone api already, it is a better idea for user to be able to import targets not only from bookmarked ones but we should let them search the target and import.
For example while adding the target we can have a search box that will search and import targets, we should let user's which targets to import."

Yeah great idea. But instead of targets, perhaps we perform this search at the program\organization level ? Searching  all the scopes in H1 and in the the future other programs might be difficult depending on the API interface of the bug bounty programs.

**Comment by yogeshojha** [comment]
@null-ref-0000 Look how excellent this PR has turned out! ;p

image

**Comment by null-ref-0000** [comment]
Wow, looks amazing!

**Comment by yogeshojha** [comment]
Hey @null-ref-0000 this turned out to be more complex than I thought;p

Check this out. If you have time, please test this out and let me know. But I want to let you know that this is a game-changing feature. Kudos to the submission!!!


https://github.com/user-attachments/assets/055fabd8-ffcc-4d01-9fd0-7d6a79cff3b7

**Comment by yogeshojha** [comment]
@null-ref-0000 this is now ready and turned out really excellent. 

If you can test it out and let me know otherwise check these videos out, what do you think?

For Importing Programs

https://github.com/user-attachments/assets/c149b674-8661-44fb-a9f2-ae8a5b8e7341



For syncing Bookmarked Programs


https://github.com/user-attachments/assets/dcdfe694-9cd6-40e4-b85e-11ec7feb7455


If everything is good, I will go ahead and merge, in next release I will target for intigriti, checked out their API looks good!

**Comment by yogeshojha** [comment]
@null-ref-0000 I have merged this in the release/2.2.0 branch

If you wish to test please test it out on release/2.2.0 and let me know. 

Once again, thank you for this awesome feature request as well as initiating this PR! A huge thanks 🚀

**Comment by null-ref-0000** [comment]
@yogeshojha I've started testing, I noticed the sync process is adding out of scope assets from H1. Also its not adding wildcard assets. I also noticed we don't set a description on the organizations, should we indicate something like synced from h1 bookmarks or imported from h1 ?

**Comment by yogeshojha** [comment]
@null-ref-0000 that's an interesting observation, do you know what keys are set for out of scope? Also for description yes I will add that.

Overall ui and performance what do you think of it?

**Comment by yogeshojha** [comment]
@null-ref-0000 Yes Intrigiti is super simple to implement as well, I checked it out yesterday. I will target for 2.3 next month for intigriti.

**Comment by null-ref-0000** [comment]
Over all UI performance is excellent, the touches you have added for notifications make for a good user experience. 


"eligible_for_submission": false is the key for out of scope assets.

**Comment by null-ref-0000** [comment]
If you looking for a program of the wild card issue, Sephora is a good public program. In this case it has one wildcard, and it doesn't even import because of the issue. In my orginal code I stripped out https:?/*. or  *. from the wild card assets. I think its important that we support this asset type.

**Comment by yogeshojha** [comment]
@null-ref-0000 I have fixed them up, however

1. sync process is not removing unbookmarked programs: this one will require a lot of work, plus we also need to provide user if they want to remove the program that are unbookmarked, so lets target this for another release, I have noted this already, will be creating feature request

2. its not adding wildcard assets: now its supported, however *.* we cant identify main domain, so it doesn't import, otherwise I checked for domains like *.hello.com its importing successfully

3. I also noticed we don't set a description on the organizations, should we indicate something like synced from h1 bookmarks or imported from h1 ?: this has been added

image

4. For out of scope domains, this has been fixed as well

If you want to test out, fetch the latest 2.2.0 branch there all these changes have been merged.

**Comment by yogeshojha** [comment]
@null-ref-0000 hey man what's your twitter id? since I will be posting this release update on twitter, I would like to give you a shoutout!

**Comment by null-ref-0000** [comment]
on twitter @null_ref_0000

**Comment by yogeshojha** [comment]
Hey @null-ref-0000 am I missing something?

**Pull request merged**